SpeechToText (#267)
MicrosoftDocs/CommunityToolkit: docs/maui/essentials/speech-to-text.md

---
title: SpeechToText - .NET MAUI Community Toolkit
author: VladislavAntonyuk
description: The SpeechToText provides the ability to convert speech to text.

> ms.date: 12/11/2022

Reviewer: ```suggestion
ms.date: 06/05/2023
```

> ms.date: 06/05/2023

Reviewer: ```suggestion
ms.date: 26/05/2023
```

---

# SpeechToText

> The `SpeechToText` provides the ability to convert speech to text.

Reviewer: ```suggestion
The `SpeechToText` API provides the ability to convert speech to text.
```

> ![Screenshot of an SpeechTexy on macOS](../images/essentials/speech-to-text-mac.gif "SpeechToText on macOS")

Reviewer: ```suggestion
![Screenshot of SpeechText implemented on macOS](../images/essentials/speech-to-text-mac.gif "SpeechToText on macOS")
```

The following preconditions required for the `SpeechToText`:
# [Android](#tab/android)

Add permissions to `AndroidManifest.xml`:

```xml
<uses-permission android:name="android.permission.RECORD_AUDIO" />
```

# [iOS/MacCatalyst](#tab/ios)

Add permissions to `Info.plist`

```xml
<key>NSSpeechRecognitionUsageDescription</key>  
<string>SpeechToText requires speech recognition usage</string>  
<key>NSMicrophoneUsageDescription</key>  

> <string>SpeechtToText requires microphone usage</string> 

Reviewer: ```suggestion
<string>SpeechToText requires microphone usage</string> 
```
